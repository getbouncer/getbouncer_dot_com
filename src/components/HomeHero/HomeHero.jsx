import React from 'react'
import PropTypes from 'prop-types'
import { graphql, useStaticQuery } from 'gatsby'
import Container from '~globals/Container'
import CtaButton from '~components/CtaButton'
import CompaniesLogos from '~components/CompaniesLogos'
import { AboveSmartphone, Smartphone } from '~components/Media'
import {
  ContentContainer,
  Headline,
  Description,
  LeftIllustration,
  RightIllustration,
  LogosContainer,
  Wrapper,
} from './styles'

export const PureHomeHero = ({ data, logos }) => {
  const {
    hero_cta: heroCta,
    hero_description: heroDescription,
    hero_illustration_left: heroIllustrationLeft,
    hero_illustration_right: heroIllustrationRight,
    hero_title: heroTitle,
  } = data

  return (
    <Wrapper>
      <Container isLarge>
        <AboveSmartphone>
          <LeftIllustration src={heroIllustrationLeft.url} alt={heroIllustrationLeft.alt} />
          <RightIllustration src={heroIllustrationRight.url} alt={heroIllustrationRight.alt} />
        </AboveSmartphone>
        <ContentContainer>
          <Headline dangerouslySetInnerHTML={{ __html: heroTitle.html }} />
          <Description dangerouslySetInnerHTML={{ __html: heroDescription.html }} />
          <div>
            <CtaButton isInternal to="/contact" text={heroCta} />
          </div>
          <Smartphone>
            <LeftIllustration src={heroIllustrationLeft.url} alt={heroIllustrationLeft.alt} />
          </Smartphone>
        </ContentContainer>
        <LogosContainer>
          <CompaniesLogos logos={logos} />
        </LogosContainer>
      </Container>
    </Wrapper>
  )
}

PureHomeHero.propTypes = {
  data: PropTypes.shape().isRequired,
  logos: PropTypes.shape().isRequired,
}

export const HomeHero = ({ data }) => {
  const logos = useStaticQuery(graphql`
    query LogosQuery {
      prismicGlobals {
        data {
          companies_logo {
            company_logo {
              alt
              url
            }
          }
        }
      }
    }
  `)

  return <PureHomeHero data={data} logos={logos.prismicGlobals.data} />
}

HomeHero.propTypes = {
  data: PropTypes.shape().isRequired,
}

export default HomeHero
