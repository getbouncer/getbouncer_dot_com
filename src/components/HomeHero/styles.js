import styled from '@emotion/styled'
import { breakpoints, theme } from '~styles'

export const Wrapper = styled.section`
  padding-bottom: 100px;

  @media ${breakpoints.smartphone} {
    padding-top: 50px;
  }
`

export const ContentContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-height: 80vh;
  text-align: center;
`

export const Headline = styled.div`
  margin: 0 auto;
  max-width: 600px;

  strong {
    background: ${theme.textGradient};
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  }
`

export const Description = styled.div`
  margin: 24px auto;
  width: 32vw;
  max-width: 550px;

  @media ${breakpoints.smartphone} {
    width: 100%;
    max-width: 310px;
  }
`

export const LeftIllustration = styled.img`
  bottom: 0;
  left: -3%;
  margin: auto;
  min-width: 26%;
  position: absolute;
  top: 0;
  width: calc(100% - 943px);

  @media ${breakpoints.smartphone} {
    position: static;
    width: 100%;
    max-width: 300px;
    margin: 45px auto;
  }
`

export const RightIllustration = styled.img`
  top: 0;
  bottom: 0;
  transform: translateY(80px);
  margin: auto;
  min-width: 36%;
  position: absolute;
  right: -5vh;
  width: calc(100% - 793px);
`

export const LogosContainer = styled.div`
  margin: 100px auto 0;
  max-width: ${theme.maxWidth};

  @media ${breakpoints.smartphone} {
    width: 100%;
    transform: none;
    position: static;
  }
`
