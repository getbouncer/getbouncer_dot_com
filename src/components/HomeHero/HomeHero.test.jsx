import React from 'react'
import renderer from 'react-test-renderer'
import { describe, expect, it } from 'jest'
import { PureHomeHero as HomeHero } from './HomeHero'

describe('HomeHero', () => {
  it('renders correctly', () => {
    const data = {
      hero_cta: 'hero cta',
      hero_description: {
        html: '<p>hero description</p>',
      },
      hero_illustration_left: {
        alt: 'hero illustration left',
        url: 'https://www.example.com/hero_illustration_left.png',
      },
      hero_illustration_right: {
        alt: 'hero illustration right',
        url: 'https://www.example.com/hero_illustration_right.png',
      },
      hero_title: {
        html: '<h1>hero title</h1>',
      },
    }
    const logos = {
      companies_logo: [
        {
          company_logo: {
            alt: 'logo 1',
            url: 'https://www.example.com/logo_1.svg',
          },
        },
        {
          company_logo: {
            alt: 'logo 2',
            url: 'https://www.example.com/logo_2.svg',
          },
        },
      ],
    }
    const tree = renderer
      .create(<HomeHero data={data} logos={logos} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
