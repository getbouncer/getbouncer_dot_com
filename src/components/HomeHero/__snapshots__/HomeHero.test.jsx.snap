// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeHero renders correctly 1`] = `
<section
  className="css-kwenbi"
>
  <div
    className="css-sdcyqc"
  >
    <div
      className="css-1l4bxlm"
    >
      <div
        className="css-m90dr8"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<h1>hero title</h1>",
          }
        }
      />
      <div
        className="css-1wdsh1s"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>hero description</p>",
          }
        }
      />
      <div>
        <a
          className="css-e1lh64"
          href="/contact"
          inverted={false}
        >
          hero cta
        </a>
      </div>
    </div>
    <div
      className="css-x03n0l"
    >
      <ul
        className="css-i0unku"
      >
        <li
          className="css-ecly76"
        >
          <img
            alt="logo 1"
            src="https://www.example.com/logo_1.svg"
          />
        </li>
        <li
          className="css-ecly76"
        >
          <img
            alt="logo 2"
            src="https://www.example.com/logo_2.svg"
          />
        </li>
      </ul>
    </div>
  </div>
</section>
`;
